### PR TITLE
Move DNS update script into template

### PR DESCRIPTION
**Pull request by nicolasbock**
Inlining the script leads to an "IndentationError: unexpected indent"
error. Since it's a larger script, we might as well place it into a
template.

Signed-off-by: Nicolas Bock <nicolas.bock@suse.com>

### CODE REVIEW

**Comment by nicolasbock** [comment]
I ran this on a freshly installed SP4. I don't know whether this issue is a new one (maybe the relatively new Ansible 2.8?), but I figured it can't hurt to move the script into a separate file.

**Comment by nicolasbock** [comment]
Another note. `prepare_infra` works without this patch if I require `ansible==2.6.16` in the venv. So this looks like some Ansible version thing.

**Comment by nicolasbock** [comment]
Ah, great suggestion! I'll change that in the next iteration :smile:

**Comment by spacefito** [comment]
Maybe too much work, but is this a candidate for an ansible module? 
It could be as easy as copy and paste. and you get better integration with ansible.

File: scripts/jenkins/ardana/ansible/roles/heat-generator/files/dns-update.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        default=[])
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    options = parse_commandline()
+    print(options)
+
+    with open('cloudConfig.yml') as f:
+        data = yaml.load(f.read(), Loader=yaml.SafeLoader)
```
Comment (spacefito) [inline scripts/jenkins/ardana/ansible/roles/heat-generator/files/dns-update.py]: can we leave this as an absolute path?  i.e. where is the file being opened.  Maybe add a new argument: the location of cloudConfig.yml as a full path, and add a default value to cover the simple case

Comment (nicolasbock) [reply scripts/jenkins/ardana/ansible/roles/heat-generator/files/dns-update.py]: That's a good idea! I added a command line option.

**Comment by nicolasbock** [comment]
I'll have a look at the Ansible module suggestion @spacefito .

**Comment by nicolasbock** [comment]
@spacefito Please have another look. I moved the script into a module.

**Comment by nicolasbock** [comment]
Yes, I tried it a few times. It should be good to go.

**Pull request merged**